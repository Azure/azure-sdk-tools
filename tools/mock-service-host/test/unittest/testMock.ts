/* eslint-disable @typescript-eslint/no-var-requires */
import * as assert from 'assert'
import Mocker from '../../src/mid/oav/mocker'
import SwaggerMocker from '../../src/mid/oav/swaggerMocker'

describe('isManagementUrlLevel() :', () => {
    it('patch for userassignedidentities', async () => {
        const result = SwaggerMocker.mockUserAssignedIdentities(
            {
                mockAny: {
                    userAssignedIdentities: {
                        oldKey: {
                            clientId: 'oldClientId',
                            principalId: 'oldClientId'
                        }
                    }
                }
            },
            { subscriptions: 'mock' },
            false
        )

        assert.ok(
            Object.keys(result.mockAny.userAssignedIdentities).every((x) =>
                x.startsWith('/subscriptions/mock/')
            )
        )
        assert.strictEqual(Object.keys(result.mockAny.userAssignedIdentities).length, 1)
    })

    it('flattenPath', async () => {
        const result = SwaggerMocker.flattenPath(
            '/subscriptions/aaa/resourceGroups/bbb/anyType/anyValue'
        )

        assert.strictEqual(
            JSON.stringify(result),
            JSON.stringify({
                subscriptions: 'aaa',
                resourcegroups: 'bbb',
                anytype: 'anyValue'
            })
        )
    })
})

describe('Mocker: ', () => {
    it('mock base64url', async () => {
        const mocker = new Mocker()
        const result = mocker.mock(
            {
                type: 'string',
                format: 'base64url'
            },
            'fakeName'
        )

        assert.strictEqual(Buffer.from(result, 'base64url').toString('base64url'), result)
    })
})
