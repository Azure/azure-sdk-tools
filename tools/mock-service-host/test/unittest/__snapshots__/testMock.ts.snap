// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mockForExample:  mock resourceType 1`] = `
Object {
  "requestResourceId": "/subscriptions/a9cbceed-cd54-47f1-b7a2-bad149218603/resourceGroups/mockRg/providers/Microsoft.Resources/mockResource/mockName",
  "requestResourceType": "Microsoft.Resources/mockResource",
  "responseResourceId": "/subscriptions/xxx/resourceGroups/yy",
  "responseResourceType": "Microsoft.Resources/mockResource",
}
`;
